toolchain/sem_ir/node_kind.h

```diff
@@ -72,6 +84,36 @@ class NodeKind : public CARBON_ENUM_BASE(NodeKind) {
 // We expect the node kind to fit compactly into 8 bits.
 static_assert(sizeof(NodeKind) == 1, "Kind objects include padding!");
 
+class NodeKind::Definition : public NodeKind {
```

Comment:
Should probably have a comment explaining how this class is used, noting @josh11b's questions about that.

Reply:
Done.